pkg/internal/request/span_test.go

```diff
@@ -21,3 +24,134 @@ func TestSpanClientServer(t *testing.T) {
 	}
 }
+
+func TestEventTypeString(t *testing.T) {
+	typeStringMap := map[EventType]string{
+		EventTypeHTTP:        "HTTP",
+		EventTypeGRPC:        "GRPC",
+		EventTypeHTTPClient:  "HTTPClient",
+		EventTypeGRPCClient:  "GRPCClient",
+		EventTypeSQLClient:   "SQLClient",
+		EventTypeRedisClient: "RedisClient",
+		EventTypeKafkaClient: "KafkaClient",
+		EventTypeRedisServer: "RedisServer",
+		EventTypeKafkaServer: "KafkaServer",
+		EventType(99):        "UNKNOWN (99)",
+	}
+
+	for ev, str := range typeStringMap {
+		assert.Equal(t, ev.String(), str)
+	}
+}
+
+func TestIgnoreModeString(t *testing.T) {
+	modeStringMap := map[IgnoreMode]string{
+		IgnoreMetrics:  "Metrics",
+		IgnoreTraces:   "Traces",
+		IgnoreMode(0):  "(none)",
+		IgnoreMode(99): "UNKNOWN (99)",
+	}
+
+	for mode, str := range modeStringMap {
+		assert.Equal(t, mode.String(), str)
+	}
+}
+
+func TestKindString(t *testing.T) {
+	m := map[*Span]string{
+		&Span{Type: EventTypeHTTP}:                                  kServer,
+		&Span{Type: EventTypeGRPC}:                                  kServer,
+		&Span{Type: EventTypeKafkaServer}:                           kServer,
+		&Span{Type: EventTypeRedisServer}:                           kServer,
+		&Span{Type: EventTypeHTTPClient}:                            kClient,
+		&Span{Type: EventTypeGRPCClient}:                            kClient,
+		&Span{Type: EventTypeSQLClient}:                             kClient,
+		&Span{Type: EventTypeRedisClient}:                           kClient,
+		&Span{Type: EventTypeKafkaClient, Method: MessagingPublish}: kProducer,
+		&Span{Type: EventTypeKafkaClient, Method: MessagingProcess}: kConsumer,
+		&Span{}: kInternal,
+	}
+
+	for span, str := range m {
+		assert.Equal(t, kindString(span), str)
+	}
+}
+
+type jsonObject = map[string]interface{}
+
+func deserializeJSONObject(data []byte) (jsonObject, error) {
+	var object jsonObject
+	err := json.Unmarshal(data, &object)
+
+	return object, err
+}
+
+func getObjFromObj(obj jsonObject, key string) jsonObject {
+	return obj[key].(jsonObject)
+}
+
+func TestSerializeJSONSpans(t *testing.T) {
+	span := Span{
+		Type:           EventTypeHTTP,
+		IgnoreSpan:     IgnoreMetrics,
+		Method:         "method",
+		Path:           "path",
+		Route:          "route",
+		Peer:           "peer",
+		PeerPort:       1234,
+		Host:           "host",
+		HostPort:       5678,
+		Status:         200,
+		ContentLength:  1024,
+		RequestStart:   10000,
+		Start:          15000,
+		End:            35000,
+		TraceID:        trace2.TraceID{0x1, 0x2, 0x3},
+		SpanID:         trace2.SpanID{0x1, 0x2, 0x3},
+		ParentSpanID:   trace2.SpanID{0x1, 0x2, 0x3},
+		Flags:          1,
+		PeerName:       "peername",
+		HostName:       "hostname",
+		OtherNamespace: "otherns",
+		Statement:      "statement",
+	}
+
+	data, err := json.MarshalIndent(span, "", " ")
+
+	require.NoError(t, err)
+
+	s, err := deserializeJSONObject(data)
+
+	require.NoError(t, err)
+
+	assert.Equal(t, s["type"], "HTTP")
```

Comment:
I think we can make this simpler by checking in one go, we have examples throughout the code, for example:
```
	assert.Equal(t, map[string]map[string]string{
		"/user/1234": {
			string(semconv.ServiceNameKey):      "svc-1",
			string(semconv.ServiceNamespaceKey): "ns",
			string(attr.ClientAddr):             "1.1.1.1",
			string(attr.HTTPRequestMethod):      "GET",
			string(attr.HTTPResponseStatusCode): "201",
			string(attr.HTTPUrlPath):            "/user/1234",
			string(attr.ServerPort):             "8080",
			string(attr.ServerAddr):             events["/user/1234"]["server.address"],
		},
		"/user/4321": {
			string(semconv.ServiceNameKey):      "svc-3",
			string(semconv.ServiceNamespaceKey): "ns",
			string(attr.ClientAddr):             "1.1.1.1",
			string(attr.HTTPRequestMethod):      "GET",
			string(attr.HTTPResponseStatusCode): "203",
			string(attr.HTTPUrlPath):            "/user/4321",
			string(attr.ServerPort):             "8080",
			string(attr.ServerAddr):             events["/user/1234"]["server.address"],
		},
	}, events)
```